### PR TITLE
New filtering experience UI enhancements

### PR DESCRIPTION
**Pull request by samwinslow**
## Changes

Contributes to #4050:

-  Filtering experience. Local filters should be collapsed by default (opened through filter icon at the right of each row). Add more filters by clicking a plus button (see wireframes).
- UI of filter rows, zebra-striped to clearly distinguish each row (as shown on mockup).
- Change filter icon to "X" to avoid confusion with removing entire series.

Screen Shot 2021-05-11 at 6 42 57 AM

- Missing icons on display config (chart visualization type, and interval)

Screen Shot 2021-05-11 at 6 43 29 AM
Screen Shot 2021-05-11 at 6 43 37 AM

- Remove event popover for PostHog-based events (as we did for properties).
- Funnels. We'll probably want to keep vertical layout for them (due to the nature of the steps process, plus there's no benefit of how these are read horizontally
- (nice to have). Tabbed keyboard navigation.

## Checklist

- [ ] All querysets/queries filter by Organization, by Team, and by User
- [ ] Django backend tests
- [x] Jest frontend tests
- [x] Cypress end-to-end tests
- [ ] Migrations are safe to run at scale (e.g. PostHog Cloud) – present proof if not obvious

### CODE REVIEW

**Comment by samwinslow** [comment]
Tests failing inconsistently / appear unrelated. Different tests fail each time.

File: frontend/src/scenes/insights/InsightTabs/InsightDisplayConfig.tsx [hunk 1]
```diff
@@ -23,7 +23,7 @@ export function InsightDisplayConfig({
     horizontalUI,
     ...props
 }: InsightDisplayConfigProps & { horizontalUI: boolean }): JSX.Element {
-    return horizontalUI ? (
+    return horizontalUI && props.activeView !== ViewType.FUNNELS ? (
```
Comment (paolodamico) [inline frontend/src/scenes/insights/InsightTabs/InsightDisplayConfig.tsx:26]: I think this extra check is not needed as you already check for the funnels view when passing down `horizontalUI`

**Review by paolodamico — APPROVED** [review]
Loving this! The new filtering button/experience is amazing and feels so fast! Also, great stuff on the display icons, wasn't picturing them that way, love them! 

Some minor things, but nothing blocking (numbered for easy discussion):
1. The zebra-striped is separating each filter but I think to be more useful it should be for each _graph series group_ (series definition + filters) that's separated. That way if you have multiple filters on each series and multiple series, you can quickly identify which one is which.
2. Arrow is not centered vertically.

3. Genuine question. Should we use the same tag indicator we use elsewhere for the "AND"s? Not sure if actually just plain text would be clearer here.

4. For the interval filters, I would just add the clock icon to the main label display (instead of on each option). Like we have for the date range selector. My thinking is that the icons don't really provide extra value.
5. (nice to have). With the introduction of tabbed navigation, would be pretty useful to be able to close the events and properties selectors with the "Esc" key.

**Comment by samwinslow** [comment]
@paolodamico for some reason I can't reply inline to your review — could you clarify this?

> Actually it looks like the display selector is no longer visible on funnels. (`frontend/src/lib/components/ChartFilter/ChartFilter.js`)

Do you mean _it's okay_ that the selector is no longer visible on funnels? In such a case, should we then remove this block on line 53 and just keep the second part of the ternary?

```
const options =
        filters.insight === ViewType.FUNNELS
            ? [
// ...
```

> 2. Arrow is not centered vertically.

Also can't repro this — how does it look for you here on Insights when the feature flag is turned off? I suspect the problem might be because the arrow is actually a Unicode character; perhaps different browsers or OSX versions use a different system symbols font.

Screen Shot 2021-05-11 at 1 51 04 PM

3. Yes, I like the `AND` tags and would be happy to use them! Related to point 2, should we keep this arrow thing at all? Perhaps an indent before "where" is all we need.

EDIT: New approach re: point 3:

Screen Shot 2021-05-11 at 3 44 34 PM

**Comment by paolodamico** [comment]
Np, thanks for the quick turnaround here! 
1. I actually think it should be there, because we use that to switch between plain funnel and funnel conversion rate. 
2. Yeah that might be it, maybe better to use Ant's. Also from your screenshot not sure if you're on the latest version of the branch (as we dropped the Add filters button). 
3. Yup might be a good option, we could try it out. Regardless, I would at most keep it for the first filter row of each series.

**Comment by samwinslow** [comment]
Cool, thanks! On issue 2 above (arrow styling) — that screenshot is what it looks like for me on master or with the feature flag turned off. Wondering if it looks the same for you or is also messed up. If so I'll switch the whole thing over to a different SVG icon.

**Comment by paolodamico** [comment]
If it's okay, I'll merge this in and we can figure out these small details in a subsequent PR, as I've made a couple of follow-up PRs from this branch and so we can wrap all these for tomorrow.

**Comment by samwinslow** [comment]
I am genuinely quite `:confused:` as to how you got the icon to display on only the select toggle... here's my attempt to do the same. Not bad necessarily, but not what was intended.

Screen Shot 2021-05-12 at 9 56 40 AM

**Comment by paolodamico** [comment]
Yeah I relied on the `optionLabelProp` of Ant's `Select`, but can't seem to reproduce for the interval filter either. Perhaps it's just a matter of having the icon separate from the component? We just need to make sure it triggers the dropdown. Are you working on this in another PR? Happy to take a look.

**Pull request merged**